plugins/chat/app/models/chat/reviewable_message.rb

```diff
@@ -117,6 +140,25 @@ def perform_delete_and_ignore(performed_by, args)
     def perform_agree_and_keep_deleted(performed_by, args)
       agree
     end
+    # TODO (reviewable-refresh): Remove combined actions above when fully migrated to new UI
+
+    def perform_no_action_message(performed_by, args)
+      if chat_message.deleted_at?
+        create_result(:success, :approved, [created_by_id], true)
+      else
+        create_result(:success, :rejected, [created_by_id], true)
+      end
+    end
+
+    def perform_restore_message(performed_by, args)
+      chat_message.trash!(performed_by)
```

Comment:
I guess `recover!` instead of `trash!` but overall actions transition look smooth  🚀

Reply:
Guardian checks are done in `Reviewable#perform`, which calls `#validate_action!` to check that the action being performed is on the list that the user has access to.